Network.hs

```diff
@@ -289,9 +290,16 @@ accept sock@(MkSocket _ AF_INET6 _ _ _) = do
          \_ -> case addr of
                  SockAddrInet  _   a   -> inet_ntoa a
                  SockAddrInet6 _ _ a _ -> return (show a)
-# if !defined(mingw32_HOST_OS)
+#if defined(mingw32_HOST_OS)
+                 SockAddrUnix {}       -> error "accept: socket address not supported on this platform."
+#else
                  SockAddrUnix      a   -> return a
-# endif
+#endif
+#if defined(CAN_SOCKET_SUPPORT)
+                 SockAddrCan {}        -> error $ "accept: unsupported for CAN peer."
+#else
+                 SockAddrCan {}        -> error "accept: socket address not supported on this platform."
+#endif
```

Comment:
We probably want `throwIO` here as well. After that I'm happy to merge this. Thanks!